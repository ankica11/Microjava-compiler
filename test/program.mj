program MJProgram

    
   int a, b, mika;
   int anka;
   int niz[];
   int niz2[];
   const int size=7;
   const char karakter='A';
   char m;
   
	
{

	
	
	void main() bool flag;
	 {
          mika=m;
	  read(b);
	  a=b+10;
	  b--;
	  print(a);
	  print(b ? 'A' : 'B');

	    
	}
	
	
	
	
}
