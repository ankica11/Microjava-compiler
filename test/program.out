
  pos: instruction operands
     | expressionstack
-----------------------------
    0: enter       0 1 
     | 
    3: read        
     | 1 
    4: putstatic   1 
     | 
    7: getstatic   1 
     | 1 
   10: const       10 
     | 1 10 
   15: add         
     | 11 
   16: putstatic   0 
     | 
   19: getstatic   1 
     | 1 
   22: const_1     
     | 1 1 
   23: sub         
     | 0 
   24: putstatic   1 
     | 
   27: getstatic   0 
     | 11 
   30: const_5     
     | 11 5 
   31: print          11
     | 
   32: getstatic   1 
     | 0 
   35: const       65 
     | 0 65 
   40: const       66 
     | 0 65 66 
   45: ???         
     | 66 0 65 66 
   46: pop         
     | 66 0 65 
   47: ???         
     | 65 66 0 65 
   48: pop         
     | 65 66 0 
   49: const_0     
     | 65 66 0 0 
   50: jne         9 
     | 65 66 
   53: ???         
     | 66 65 66 
   54: pop         
     | 66 65 
   55: pop         
     | 66 
   56: jmp         4 
     | 66 
   60: const_1     
     | 66 1 
   61: bprint      B
     | 
   62: exit        
     | 
   63: return      
Completion took 4615 ms